src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -37,24 +62,26 @@ end
 dimension(::ExtendedHubbardReal1D, address) = number_conserving_dimension(address)
 
 LOStructure(::Type{<:ExtendedHubbardReal1D{<:Real}}) = IsHermitian()
+LOStructure(::Type{<:ExtendedHubbardReal1D{<:Complex}}) = IsHermitian()
```

Comment:
This is not true:
```julia
julia> addr = BoseFS((1,1,1,1)) # can be whatever
julia> H = ExtendedHubbardReal1D(addr; t=im)
julia> LOStructure(H)
IsHermitian()
julia> ishermitian(Matrix(H))
false
```

@joachimbrand does it make sense to support complex hopping?

Reply:
I think there is a problem with the sign in the complex term of t, which is not being taken care of concerning the direction of hopping. This leads to the situation H != H'.

Reply:
julia> addr = BoseFS((1,1,1,1)) # can be whatever
BoseFS{4,4}(1, 1, 1, 1)

julia> H = ExtendedHubbardReal1D(addr; t=im)
ExtendedHubbardReal1D(BoseFS{4,4}(1, 1, 1, 1); u=1.0 + 0.0im, v=1.0 + 0.0im, t=0.0 + 1.0im, boundary_condition=:periodic)

julia> ishermitian(Matrix(H))
true

julia> ishermitian(Matrix(H))
true

@mtsch  I have taken care of this problem.

Reply:
Ah, actually there are different cases. Complex hopping but with a hermitian Hamiltonian (as implemented by @Skuwar1 ) makes sense. It also may make sense to support complex `u` or `v`. However, in this case the Hamiltonian is not hermitian any more. So `LOStructure` should actually check the imaginary parts of `u` and `v` and return `IsHermitian()` only if both are zero.